src/vs/workbench/browser/web.simpleservices.ts

```diff
@@ -809,7 +805,7 @@ export class SimpleWindowService implements IWindowService {
 			try {
 				if (!(<any>document).webkitIsFullScreen) {
 					(<any>target).webkitRequestFullscreen(); // it's async, but doesn't return a real promise.
-					browser.setFullscreen(true);
+					browser.setFullscreen(true); // we have to set this proactively because Safari doesn't emit fullscreenchange event.
```

Comment:
As I commented, we can't fully rely on `fullscreenchange` as Safari doesn't implement it, even though it's Living Standard.